### PR TITLE
fix(dropdown-menu-v2): Fix ResizeObserver loop limit error

### PR DESCRIPTION
**Pull request by vuluongj20**
[Sentry Issue.](https://sentry.io/organizations/sentry/issues/2485743992/?project=11276)

ResizeObserver (added in https://github.com/getsentry/sentry/pull/32061) throws an infinite loop error when we mount `DropdownMenuControlV2`. This only happens when the component is first mounted, not in subsequent updates. The loop may have been caused by `DropdownButton` rendering multiple times at different widths during mount. 

**Solution:** wrapping the contents of `updateTriggerWidth` inside a `setTimeout` seems to fix the issue. Doing this ensures that `updateTriggerWidth ` only runs once the trigger has finished rendering.

**Note: To see the infinite loop error in the console,** we need to add
```js
window.onerror = console.log;
```
to the code (anywhere is fine). The error isn't logged by default.

### CODE REVIEW

File: static/app/components/dropdownMenuControlV2.tsx [hunk 1]
```diff
@@ -127,8 +127,10 @@ function MenuControl({
   const [triggerWidth, setTriggerWidth] = useState<number>();
   // Update triggerWidth when its size changes using useResizeObserver
   const updateTriggerWidth = useCallback(() => {
-    const newTriggerWidth = ref.current?.offsetWidth;
-    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+    setTimeout(() => {
+      const newTriggerWidth = ref.current?.offsetWidth;
+      !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+    });
```
Comment (evanpurkhiser) [inline static/app/components/dropdownMenuControlV2.tsx:133]: Let's definitely add a comment.

Also I might write this as 


```tsx
  const updateTriggerWidth = useCallback(async() => {
		// XXX: Comment why we do this
    await new Promise(resolve => setTimeout(resolve));

    const newTriggerWidth = ref.current?.offsetWidth;
    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
  })
```

File: static/app/components/dropdownMenuControlV2.tsx [hunk 2]
```diff
@@ -127,8 +127,10 @@ function MenuControl({
   const [triggerWidth, setTriggerWidth] = useState<number>();
   // Update triggerWidth when its size changes using useResizeObserver
   const updateTriggerWidth = useCallback(() => {
-    const newTriggerWidth = ref.current?.offsetWidth;
-    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+    setTimeout(() => {
```
Comment (dashed) [inline static/app/components/dropdownMenuControlV2.tsx:130]: @vuluongj20 I'm wondering if `requestAnimationFrame()` is a modern alternative as suggested in https://github.com/adobe/react-spectrum/issues/1924#issuecomment-871623619

File: static/app/components/dropdownMenuControlV2.tsx (hunk 1 above)
Comment (vuluongj20) [reply static/app/components/dropdownMenuControlV2.tsx:133]: hmm what's the benefit of using a promise over `setTimeout` or @dashed's suggestion `requestAnimationFrame`?

Comment (evanpurkhiser) [reply static/app/components/dropdownMenuControlV2.tsx:133]: Less indentation

**Review by dashed — APPROVED** [review]

**Pull request merged**